client/components/search/index.jsx

```diff
@@ -201,15 +198,11 @@ const Search = React.createClass( {
 	},
 
 	keyUp: function( event ) {
-		if ( event.which === 13 && isMobile() ) {
+		if ( event.key === 'Enter' && isMobile() ) {
```

Comment:
>keyboard events are so fun.

So much! I dug around a bit and, yes React does a whole bunch of normalization. 

The **TL;DR** on below is, when using keyup/keydown, ```event.key``` is reliable for non character keys such as ```Enter``` and ```ArrowRight```, but detection of characters is best left to ```event.keyCode```.

First, ```event.which``` [falls back to ```event.keyCode```](https://github.com/facebook/react/blob/v0.14.3/src/renderers/dom/client/syntheticEvents/SyntheticKeyboardEvent.js#L66-L68) anyways, so may as well forget about it.

Next, ```event.keyCode``` is fully supported, but deprecated. Weird because there isn't a clear replacement, but its useable.

Finally, ```event.key``` is supported by all except Safari but not deprecated. React does some [normalization](https://github.com/facebook/react/blob/v0.14.3/src/renderers/dom/client/utils/getEventKey.js#L75-L85) (ie, "Esc" => "Escape") if ```event.key``` exists. Otherwise, [it maps ```event.keyCode```](https://github.com/facebook/react/blob/v0.14.3/src/renderers/dom/client/utils/getEventKey.js#L95-L99) to the string you'd expect, but only for non-character keys.

Eesh, curiosity got the best of me on this one, maybe it deserves a write-up. Anyways, I'll leave the ```event.key``` here as is.